lib/dao.js

```diff
@@ -926,13 +926,13 @@ DataAccessObject.remove = DataAccessObject.deleteAll = DataAccessObject.destroyA
 DataAccessObject.removeById = DataAccessObject.destroyById = DataAccessObject.deleteById = function deleteById(id, cb) {
   if (stillConnecting(this.getDataSource(), this, arguments)) return;
   var Model = this;
-  
-  this.remove(byIdQuery(this, id).where, function(err) {
+
+  this.getDataSource().connector.destroy(this.modelName, id, function (err) {
     if ('function' === typeof cb) {
       cb(err);
     }
     if(!err) Model.emit('deleted', id);
-  });
+  }.bind(this));
```

Comment:
This change will break other things, such as the default scope. We translate deleteById to deleteAll with an id filter to make the 'delete' consistent. I understand that some connectors may choose to provide a specific implementation of destroy (by id). I need to find out what is the best way to honor that.